client/lib/rpc.rs

```diff
@@ -0,0 +1,329 @@
+        let response_value = match maybe_block_hash {
+            Some(block_hash) => {
+                let params = GetBlockParams { block_hash };
+                GetBlock::request_with_map_params(self.verbose, &node_address, self.rpc_id, params)?
+            }
+            None => GetBlock::request(self.verbose, &node_address, self.rpc_id)?,
+        };
+        Ok(serde_json::from_value::<GetBlockResult>(response_value)?)
+    }
+}
+
+/// General purpose client trait for making requests to casper-node's HTTP endpoints.
+pub(crate) trait RpcClient {
+    const RPC_METHOD: &'static str;
+
+    /// Call a casper-node JSON-RPC endpoint.
+    fn request(verbose: bool, node_address: &str, rpc_id: u32) -> Result<Value> {
+        executor::block_on(async {
+            request(
+                verbose,
+                node_address,
+                rpc_id,
+                Self::RPC_METHOD,
+                Params::None(()),
+            )
+            .await
+        })
+    }
+
+    /// Call a casper node JSON-RPC endpoint with parameters.
+    fn request_with_map_params<T: IntoJsonMap>(
+        verbose: bool,
+        node_address: &str,
+        rpc_id: u32,
+        params: T,
+    ) -> Result<Value> {
+        executor::block_on(async {
+            request(
+                verbose,
+                node_address,
+                rpc_id,
+                Self::RPC_METHOD,
+                Params::from(params.into_json_map()),
+            )
+            .await
+        })
+    }
+}
+
+// TODO - If/when https://github.com/AtsukiTak/warp-json-rpc/pull/1 is merged and published,
+//        change `rpc_id` to a `jsonrpc_lite::Id`.
+async fn request(
+    verbose: bool,
+    node_address: &str,
+    rpc_id: u32,
+    method: &str,
+    params: Params,
+) -> Result<Value> {
+    let url = format!("{}/{}", node_address, RPC_API_PATH);
+    let rpc_req = JsonRpc::request_with_params(rpc_id as i64, method, params);
+    let client = Client::new();
+    let response = client
+        .post(&url)
+        .json(&rpc_req)
+        .send()
+        .await
+        .map_err(Error::FailedToGetResponse)?;
+
+    if let Err(error) = response.error_for_status_ref() {
+        if verbose {
+            println!("Failed Sending {:?}, {}", rpc_req, error);
+        }
+        return Err(Error::FailedSending(rpc_req));
+    }
+
+    let rpc_response = response.json().await.map_err(Error::FailedToParseResponse);
+
+    if let Err(error) = rpc_response {
+        if verbose {
+            println!("Failed parsing as a JSON-RPC response: {}", error);
+        }
+        return Err(error);
+    }
+
+    let rpc_response: JsonRpc = rpc_response?;
+
+    if let Some(success) = rpc_response.get_result() {
+        if verbose {
+            println!("Received successful response:");
+            println!(
+                "{}",
+                serde_json::to_string_pretty(&rpc_response).expect("should encode to JSON")
+            );
```

Comment:
I think this should be removed as mentioned in a previous comment.  Most callers from our own binary will be printing the same output, so doing here too seems a bit over the top.

If you disagree strongly, I could be persuaded.  I can see the lack of symmetry of printing the request but not the response as being a negative.